src/main/java/com/coveo/pushapiclient/ApiCore.java

```diff
@@ -0,0 +1,51 @@
+package com.coveo.pushapiclient;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpResponse;
+
+// TODO: LENS-934 - Support throttling
+class ApiCore {
```

Comment:
Since this seems primarily a HttpClient wrapper (not sure about its plan for future) - how about `ApiCore` -> `Client` e.g. ?

Reply:
To clarify - Since the class currently is mostly a HttpClient wrapper was thinking it might be better to name it closer to its purpose. ApiCore seems to suggest more than what it currently does. But it there are future plans to add more non-httpclient-wrapper functionality, then its fine.

Reply:
Yes, I will be adding logic to handle API throttling later on.